docs/python/data-science-tutorial.md

```diff
@@ -0,0 +1,250 @@
+
+   ```python
+   import pandas as pd
+   import numpy as np
+   data = pd.read_csv('data.csv')
+   ```
+
+1. Now, run the cell using the Run cell icon.
+
+   ![Running a Jupyter notebook cell](images/data-science-tutorial/jupyter-cell-01.png)
+
+1. After the cell finishes running, you can view the data that was loaded using the variable explorer and data viewer. First click on the chart icon in the notebook's upper toolbar, then the data viewer icon to the right of the *data* variable. For additional information about the data set, refer to [this document](http://biostat.mc.vanderbilt.edu/wiki/pub/Main/DataSets/titanic3info.txt) about how it was constructed.
+
+   ![Data viewer and variable explorer](images/data-science-tutorial/data-viewer.png)
+
+   You can then use the data viewer to view, sort, and filter the rows of data. After reviewing the data, it can then be helpful to graph some aspects of it to help visualize the relationships between the different variables.
+
+1. Before the data can be graphed though, you need to make sure that there aren’t any issues with it. If you look at the Titanic csv file, one thing you’ll notice is that a question mark (“?”) was used to designate cells where data wasn’t available.
+
+   While Pandas can read this value into a DataFrame, the result for a column like Age is that its data type will be set to Object instead of a numeric data type, which is problematic for graphing.
+
+   This problem can be corrected by replacing the question mark with a missing value that pandas is able to understand. Add the following code to the next cell in your notebook to replace the question marks in the age and fare columns with the [numpy NaN](https://docs.scipy.org/doc/numpy/reference/constants.html?highlight=nan#numpy.nan) value. Notice that we also need to update the column's data type after replacing the values.
+
+   ```python
+   data.replace('?', np.nan, inplace= True)
+   data = data.astype({"age": np.float64, "fare": np.float64})
+   ```
+
+   > **Note**: If you ever need to see the data type that has been used for a column, you can use the [DataFrame dtypes](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.dtypes.html#pandas.DataFrame.dtypes) attribute.
+
+1. Now that the data is in good shape, you can use [seaborn](http://seaborn.pydata.org/) and [matplotlib](https://matplotlib.org) to view how certain columns of the dataset relate to survivability. Add the following code to the next cell in your notebook and run it to see the generated plots.
```

Comment:
Should mention if you need to make a new cell, either use the new cell button at the bottom left of the cell, or use the "B" hotkey in command mode (when the bar is blue after you press ESC)

Reply:
Note added.